dts/bindings/sensor/st,lis2mdl-i2c.yaml

```diff
@@ -17,3 +17,17 @@ properties:
         This pin defaults to active high when produced by the sensor.
         The property value should ensure the flags properly describe
         the signal that is presented to the driver.
+
+    single-mode:
+      type: boolean
+      required: false
+      description: |
+        Set to config the sensor in single measurement mode. Leave
+        unset to configure the sensor in continious measurement mode.
+
+    cancel-offset:
+      type: boolean
+      required: false
```

Comment:
Yes, If I change it to true, then it is required to be set in .dts file. So it will always be enabled which is not my intention. 

If the user obviates it, the driver instance will not cancel the offset.